arangod/Cluster/ClusterInfo.cpp

```diff
@@ -896,66 +898,80 @@ void ClusterInfo::loadPlan() {
 
         auto const collectionId = collectionPairSlice.key.copyString();
 
+        uint64_t computedHash = collectionSlice.hash();
+        uint64_t foundHash = 0;
         try {
           std::shared_ptr<LogicalCollection> newCollection;
 
+          // Try if hash is different from last time:
+          newCollection = getCollectionNTWithHash(databaseName, collectionId, foundHash);
```

Comment:
`getCollectionNTWithHash()` is called here several times in a row for the same database, and the function will thus make many repeated lookups in the `_plannedCollections` map for the same database.
This can be optimized by making a lookup for the database in `_plannedCollections` at the beginning of this loop here (we are in a loop that builds the collections for a specific database), and then only doing a lookup in the map of collections for this database.
Apart from that, `getCollectionNTWithHash` always acquires the RW-Lock on `_planProt.lock`, which technically is correct, but unnecessary. While we are in this loop that rebuilds the collections, no one else can modify `_plannedCollections`, as the only two modifying functions are `loadPlan()` and `cleanup()`, which are both protected by the same mutex. Of course it may be fragile to not acquire the RW-lock here if someone changes the code later.
But what we maybe should do is to acquire the RW-lock in read mode at the beginning of the loop and hold it for the entire loop duration. This should be safe and not block anything else (as writing `_plannedCollections` still requires grabbing the mutex).
All this would make it more safe in case of future modifications and faster, because we will safe repeated RW-lock acquistions and repeated lookups of the same database in `_plannedCollections`.

Reply:
Done.